### DISCUSSION TITLE
Invoke-PnPSiteDesign not working

### DISCUSSION THREAD

**Discussion by imsb365** [pnp/powershell#1778] [Q&A]
### Discussed in https://github.com/pnp/powershell/discussions/1688

<div type='discussions-op-text'>

<sup>Originally posted by **imsb365** March  8, 2022</sup>
I have created a custom SharePoint online site design. The design includes below actions,

> - Apply theme "My green theme"
> - Set site external sharing setting
> - Add to hub "My-Home"
> - Add "Home-Editors" to "Members"
> - Add "Home-SuperUsers" to "Owners"
> - Add "Everyone except external users" to "Visitors"
> - Update site regional settings
> - Update branding
> - Add content types from hub
> - Create or update list "Site Pages"
> - Add content type "Intranet Page"

Custom content type 'Intranet Page' is created in admin gallery and it should be synched from there to site.

I have written a PowerShell script that should apply above design to site and then the script updates some other settings in site and library that cannot be achieved from design template.

When I try to execute `Invoke-PnPSiteDesign -Identity $siteDesign.Id -WebUrl $siteUrl` the script just moves to next line without any error or warning, but when I refresh the site in browser design is not applied to SharePoint site. I tried it multiple times but still no luck.
However if I apply the design from UI menu 'Apply a site template' the design is applied.

Is there any bug in Invoke-PnPSiteDesign? or I am missing something?

PS. I have PnP.Powershell version 1.9.0
</div>

**Comment by JoeAyre** [comment]
I have the same issue. I have a Site Template (i.e. Site Design) that 'contains' two site scripts. Script one adds some site columns, and script two creates some content types, populating them with some of the site columns created via script one. When I use the Invoke-SPOSiteDesign via PowerShell, it just goes to the next line but nothing happens. If I apply the site template via the UI, both site scripts execute as expected. Pretty frustrating so I wonder if anyone can pinpoint what the problem might be?

**Comment by JoeAyre** [comment]
I think I figured it out - my site design had more than 30 actions (Invoke-SPOSiteDesign is limited to 30 actions and subactions).

Simple enough workaround:

Instead of using e.g. 

`Invoke-SPOSiteDesign -Identity xxxxxxxx-xxxx-xxxx-xxxxxxxxxxxx -WebUrl "https://<tenant>.sharepoint.com/sites/<site>"
`

...simply use...

`Add-SPOSiteDesignTask -SiteDesignId xxxxxxxx-xxxx-xxxx-xxxxxxxxxxxx -WebUrl 
 https://<tenant>.sharepoint.com/sites/<site>"`

This will schedule the site script to be applied. When you run it, you'll get given an ID for the scheduled task. You can check up on this by using:

 `Get-SPOSiteDesignTask -Identity xxxxxxxx-xxxx-xxxx-xxxxxxxxxxxx -WebUrl  https://<tenant>.sharepoint.com/sites/<site>"
`

...although for me, nothing was returned as, presumably, the task got fired off almost immediately, hence nothing 'scheduled' was there for reporting on.

**Reply by MrRCG** [reply]
This worked!
I'm using PnP but it's the same and it worked!

```
# Apply External Site Template
Add-PnPSiteDesignTask -SiteDesignId $siteDesign -WebUrl $siteUrl -Connection $siteConnection
Get-PnPSiteDesignTask -WebUrl $siteUrl -Connection $siteConnection
```

Thank-you!

**Reply by JoeAyre** [reply]
No problem - glad to help!

**Reply by mcpang** [reply]
Thank you so much, @JoeAyre , for sharing the solution. I wasted 3 days until I read your post. Microsoft couldn't figure it out. 
Thank you.
